src/appConfig.ts

```diff
@@ -0,0 +1,59 @@
+import { disabledSettings, options } from './optionsStorage'
+import { miscUiState } from './globalState'
+import { setLoadingScreenStatus } from './appStatus'
+
+export type AppConfig = {
+  // defaultHost?: string
+  // defaultHostSave?: string
+  defaultProxy?: string
+  // defaultProxySave?: string
+  // defaultVersion?: string
+  peerJsServer?: string
+  peerJsServerFallback?: string
+  promoteServers?: Array<{ ip, description, version? }>
+  mapsProvider?: string
+
+  appParams?: Record<string, any> // query string params
+
+  defaultSettings?: Record<string, any>
+  forceSettings?: Record<string, boolean>
+  // hideSettings?: Record<string, boolean>
+  allowAutoConnect?: boolean
+  pauseLinks?: Array<Array<Record<string, any>>>
+}
+
+export const loadAppConfig = (appConfig: AppConfig) => {
+  if (miscUiState.appConfig) {
+    Object.assign(miscUiState.appConfig, appConfig)
+  } else {
+    miscUiState.appConfig = appConfig
+  }
+
+  if (appConfig.forceSettings) {
+    for (const [key, value] of Object.entries(appConfig.forceSettings)) {
+      if (value) {
+        disabledSettings.value.add(key)
+        // since the setting is forced, we need to set it to that value
+        if (appConfig.defaultSettings?.[key]) {
+          options[key] = appConfig.defaultSettings[key]
+        }
+      } else {
+        disabledSettings.value.delete(key)
+      }
+    }
+  }
+}
+
+export const isBundledConfigUsed = !!process.env.INLINED_APP_CONFIG
+
+if (isBundledConfigUsed) {
+  loadAppConfig(process.env.INLINED_APP_CONFIG as AppConfig ?? {})
+} else {
+  void window.fetch('config.json').then(async res => res.json()).then(c => c, (error) => {
```

Comment:
This should probably use `await` not `void` to ensure the config was loaded before the client continues to load.

Reply:
Hm, I was avoiding any usage of top level awaits because I think it can break the app in some unexpected way and don't really see any benefit of using it

Reply:
Well I don't really see what is stopping the client from using the bundled config.json values until the remote one is loaded and imo await is the proper way to force async queries to block the app until it's loaded? If it really breaks something then that would be a bug elsewhere (and if it fails to load the file then it should of course break as the app doesn't work without the config)